lang/py/src/avro/schema.py

```diff
@@ -875,51 +875,51 @@ def __eq__(self, that):
 #
 # Module Methods
 #
-def get_other_props(all_props,reserved_props):
+def get_other_props(all_props, reserved_props):
   """
   Retrieve the non-reserved properties from a dictionary of properties
   @args reserved_props: The set of reserved properties to exclude
   """
   if callable(getattr(all_props, 'items', None)):
-    return dict([(k,v) for (k,v) in all_props.items() if k not in
-                 reserved_props ])
-
+    return {k: v for k, v in all_props.items() if k not in reserved_props}
+
+def make_bytes_decimal_schema(other_props):
+  """Make a BytesDecimalSchema from just other_props."""
+  return BytesDecimalSchema(other_props.get('precision'), other_props.get('scale', 0))
+
+def make_logical_schema(logical_type, type_, other_props):
+  """Map the logical types to the appropriate literal type and schema class."""
+  logical_types = {
+    constants.DATE: ('int', DateSchema),
+    constants.DECIMAL: ('bytes', make_bytes_decimal_schema),
```

Comment:
Decimal can annotate fixed too.

Reply:
True, will revisit. Interesting -- we don't seem to have a test case for FixedDecimal that catches this.

Reply:
Hmm would you mind add one which covers this case?

Reply:
Sure. I also found that (at least in Python 2) tests written in this iterative style are counted as a single test, so like

```python
for ex in examples:
  self.assertSomethingAbout(ex)
```

is one test and if one example fails it won't continue to the other examples -- meaning you have to fix exactly one at a time. I will introduce a change so that we can generate individual tests for each example and be able to see all the failing tests in the output at once.

Reply:
Ohh, there is a test case, but it's actually a logic error that bypasses that test in a new way -- fixed is a named type, which is the only named type that can be a logical type.

Reply:
No, strike that, I'm just tired -- it's not a logic error at all. It's actually correct. Named types are handled first, and so FixedDecimalSchema never needs to be called here. I've added a comment to the code instead to make this clear.